pkg/lib/hooks.js

```diff
@@ -324,3 +324,36 @@ export function useEvent(obj, event, handler) {
 export function useInit(func, deps, comps, destroy = null) {
     return useObject(func, destroy, deps || [], comps);
 }
+
+/* - usePolkitPermissions(actions)
+*
+ * function Component(arg) {
+ *   const permissions = usePolkitPermissions([
+ *     'org.freedesktop.NetworkManager.reload',
+ *     'org.freedesktop.NetworkManager.network-control'
+ *   ]);
+ *
+ *   ...
+ *  }
+ *
+ * The returned value from the hook is an array of booleans,
+ * in the order that they were specified, specifying if the
+ * action is allowed or not.
+ * ]);
+*/
+export function usePolkitPermissions(actions) {
+    const [permissions, setPermissions] = useState([]);
+
+    useEffect(() => {
+        Promise.allSettled(actions.map((action) =>
+            // pkcheck returns a 0 status code if the check passed
+            cockpit.spawn(['sh', '-c', `pkcheck --action-id ${action} --process $$ --allow-user-interaction 2>&1`])
+        )).then((results) => {
+            setPermissions(results.map((result) =>
+                result.status === 'fulfilled'));
```

Comment:
Returning the result as a list is IMHO rather error prone -- callers would have to remember/hardcode e.g. `results[3]` or keep a list of constants, but these would all change if you modify the checked actions. This is currently papered over as your second demo commit just uses `.every()`, but of course that's not how the final code would look like. I suggest to return the result either as an action → bool dictionary or as a set (but I think dict is better). It will look more verbose in the callers, but that also makes it easier to understand and robust.

Reply:
Fair point and I like the idea of the dict. Let me fix that up